### PR TITLE
Datastores for node graph edges

### PR DESCRIPTION
**Pull request by md2119**
## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- ~Evaluated and added CHANGELOG entry if required~
- ~Determined and documented upgrade steps~
- ~Documented user facing changes ~

If any of these don't apply, please comment below.

### CODE REVIEW

File: central/nodecomponentedge/datastore/singleton.go [hunk 1]
```diff
@@ -16,10 +20,18 @@ var (
...
 func initialize() {
-	storage := dackbox.New(globaldb.GetGlobalDackBox())
-	searcher := search.New(storage, index.New(globalindex.GetGlobalIndex()))
+	var storage store.Store
+	var searcher search.Searcher
 
-	ad = New(globaldb.GetGlobalDackBox(), storage, index.New(globalindex.GetGlobalIndex()), searcher)
+	if features.PostgresDatastore.Enabled() {
+		storage = postgres.New(globaldb.GetPostgres())
+		searcher = search.New(storage, index.New(globalindex.GetGlobalIndex()))
```
Comment (connorgorman) [inline central/nodecomponentedge/datastore/singleton.go:28]: This should be based on postgres and not on bleve

File: central/nodecomponentedge/datastore/datastore_impl.go [hunk 2]
```diff
@@ -42,38 +43,46 @@ func (ds *datastoreImpl) Count(ctx context.Context) (int, error) {
 }
 
 func (ds *datastoreImpl) Get(ctx context.Context, id string) (*storage.NodeComponentEdge, bool, error) {
-	filteredIDs, err := ds.filterReadable(ctx, []string{id})
-	if err != nil || len(filteredIDs) != 1 {
-		return nil, false, err
+	if features.PostgresDatastore.Enabled() {
```
Comment (connorgorman) [inline central/nodecomponentedge/datastore/datastore_impl.go:46]: if ! enabled?

**Review by connorgorman — CHANGES_REQUESTED** [review]
See comments

File: central/nodecomponentedge/datastore/singleton.go [hunk 3]
```diff
@@ -16,10 +20,18 @@ var (
...
+	} else {
+		storage = dackbox.New(globalDackbox.GetGlobalDackBox())
+		searcher = search.New(storage, index.New(globalindex.GetGlobalIndex()))
+	}
+
+	ad = New(globalDackbox.GetGlobalDackBox(), storage, index.New(globalindex.GetGlobalIndex()), searcher)
```
[the remark this answers is not in the document]
Comment (md2119) [reply central/nodecomponentedge/datastore/singleton.go:34]: 😱

File: central/nodecomponentedge/datastore/datastore_impl.go (hunk 2 above)
Comment (md2119) [reply central/nodecomponentedge/datastore/datastore_impl.go:46]: 😱

File: central/nodecomponentedge/datastore/singleton.go (hunk 1 above)
Comment (md2119) [reply central/nodecomponentedge/datastore/singleton.go:28]: 😱

**Review by connorgorman — APPROVED** [review]

**Review by viswajithiii — APPROVED** [review]

**Pull request merged**